File: EventBus/src/InMemoryEventBus.php
<?php

namespace EventBus;

use EventBus\Contracts\{
    EventBus,
    Listener,
    Event
};

class InMemoryEventBus implements EventBus {

    private array $listeners = [];

    public function subscribe(Listener $listener): self
    {
        $this->listeners[] = $listener;

        return $this;
    }

    public function publish(Event $event): self
    {
        foreach ($this->listeners as $listener) {
            $listener->notify($event);
        }

        return $this;
    }
}

File: EventBusC/InMemoryEventBus.cs
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Threading.Tasks;
using EventBusC.Contracts;

namespace EventBusC
{
    public class InMemoryEventBus : IEventBus
    {
        public Task PublishAsync<TEvent>(TEvent @event) where TEvent : IEvent
        {

        }

        public void Subscribe<TEvent>(IListener<TEvent> listener) where TEvent : IEvent
        {

        }
    }
}
